// fay-react-lib/user.js
import _objectSpread from "@babel/runtime-corejs2/helpers/objectSpread";

/**
 * Created by feichongzheng on 17/9/28.
 */
import cookie from 'react-cookie';
var baseOption = {
  path: '/'
};

var loginUser = function loginUser() {
  return cookie.load('current-user');
};

var isLogin = function isLogin() {
  var user = loginUser();
  return typeof user === 'object';
};

var logout = function logout(history, location) {
  cookie.remove('current-user', baseOption);
  history.push('/login', {
    from: location
  });
};

var goToLogin = function goToLogin(history, location) {
  cookie.remove('current-user');
  history.push('/login', {
    from: location
  });
};

var saveUser = function saveUser(value, rememberTime) {
  var opt = rememberTime ? _objectSpread({}, baseOption, {
    maxAge: rememberTime,
    expires: new Date(new Date().getTime() + rememberTime * 1000)
  }) : baseOption;
  cookie.save('current-user', value, opt);
};

var saveUserLoginInfo = function saveUserLoginInfo(value, rememberTime) {
  var baseOption = {
    path: '/'
  };
  var opt = rememberTime ? _objectSpread({}, baseOption, {
    maxAge: rememberTime,
    expires: new Date(new Date().getTime() + rememberTime * 1000)
  }) : baseOption;
  cookie.save('login-info', value, opt);
};

var removeUserLoginInfo = function removeUserLoginInfo() {
  cookie.remove('login-info');
};

var removeUser = function removeUser() {
  cookie.remove('current-user');
};

export { loginUser, isLogin, logout, goToLogin, saveUser, removeUser, saveUserLoginInfo, removeUserLoginInfo };
