// fay-react-lib/loadable.js
import _inheritsLoose from "@babel/runtime-corejs2/helpers/inheritsLoose";
import React from 'react';
import Loadable from 'react-loadable';
import LoadableBundle from './loadableBundle';
import ErrorBoundary from './error/errorBoundary';

var LoadableMap =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(LoadableMap, _React$Component);

  function LoadableMap(props) {
    var _this;

    _this = _React$Component.call(this, props) || this;
    var mod = Loadable.Map({
      loader: props.loader,
      render: function render(loaded, props) {
        var view = loaded.view,
            reducer = loaded.reducer,
            sagas = loaded.sagas;
        view = view && view.default;
        reducer = reducer && reducer.default;
        sagas = sagas && sagas.default;

        if (reducer || sagas) {
          return React.createElement(ErrorBoundary, null, React.createElement(LoadableBundle, {
            reducer: reducer,
            sagas: sagas,
            view: view,
            viewProps: props
          }));
        } else {
          return React.createElement(ErrorBoundary, null, React.createElement(view, props));
        }
      },
      loading: function loading() {
        return React.createElement("div", null, "\u9875\u9762\u52A0\u8F7D\u4E2D...");
      }
    });
    _this.state = {
      mod: mod
    };
    return _this;
  }

  var _proto = LoadableMap.prototype;

  _proto.render = function render() {
    var View = this.state.mod;
    return React.createElement(View, this.props.props);
  };

  return LoadableMap;
}(React.Component);

export default LoadableMap;
