// fay-react-lib/loadableBundle.js
import _inheritsLoose from "@babel/runtime-corejs2/helpers/inheritsLoose";
import _defineProperty from "@babel/runtime-corejs2/helpers/defineProperty";

/**
 * Created by feichongzheng on 17/9/12.
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { injectAsyncStore, rejectAsyncStore } from './store';

var Bundle =
/*#__PURE__*/
function (_Component) {
  _inheritsLoose(Bundle, _Component);

  function Bundle() {
    return _Component.apply(this, arguments) || this;
  }

  var _proto = Bundle.prototype;

  _proto.componentWillUnmount = function componentWillUnmount() {
    if (this.props.cache === false) rejectAsyncStore(this.props.reducer);
  };

  _proto.componentWillMount = function componentWillMount() {
    var _this$props = this.props,
        reducer = _this$props.reducer,
        sagas = _this$props.sagas;
    injectAsyncStore(this.context.store, reducer, sagas);
  };

  _proto.render = function render() {
    var viewProps = this.props.viewProps;
    var View = this.props.view;
    return React.createElement(View, viewProps);
  };

  return Bundle;
}(Component);

_defineProperty(Bundle, "propTypes", {
  cache: PropTypes.bool
});

_defineProperty(Bundle, "contextTypes", {
  store: PropTypes.object
});

export default Bundle;
