// fay-react-ui/res/business/rySelector/rySelectContent.js
import _inheritsLoose from "@babel/runtime-corejs2/helpers/inheritsLoose";
import _assertThisInitialized from "@babel/runtime-corejs2/helpers/assertThisInitialized";
import _getIterator from "@babel/runtime-corejs2/core-js/get-iterator";
import _JSON$stringify from "@babel/runtime-corejs2/core-js/json/stringify";

/**
 * Created by Administrator on 2018/5/18.
 */
import React, { Component } from "react";
import PropTypes from "prop-types";
import Row from "../../../row";
import Col from "../../../col";
import Card from "../../../card";
import Button from '../../../button';
import Search from "../../../search";
import Icon from '../../../icon';
import RyTree from "./ryTree";
import SelectedList from "./selectedList";
import QueryList from "./queryList";
import { loginUser } from "rj-lib/user";
import { userSelect } from "./api";
import "./style/index.less";
var stylePrefix = 'rj-rySelector';
var userInfo = loginUser();
var latelyContactorsSize = 20; //最多存储最近联系人个数

function onCloseSelectedList(username) {
  var selectedRys = this.state.selectedRys;
  var newSelectedRys = selectedRys.filter(function (item) {
    return item.username != username;
  });
  this.setState({
    selectedRys: newSelectedRys
  });
}

function onClickSearchClear() {
  var pagination = this.state.pagination;
  pagination.current = 1;
  pagination.pageSize = 10;
  pagination.total = 0;
  this.setState({
    tableData: [],
    pagination: pagination,
    searchValue: ''
  });
}

function onChangeSearch(str) {
  if (str !== "") {
    fetchTableData.call(this, {
      str: str,
      current: this.state.pagination.current,
      pageSize: this.state.pagination.pageSize
    });
  } else {
    var pagination = this.state.pagination;
    pagination.current = 1;
    pagination.pageSize = 10;
    pagination.total = 0;
    this.setState({
      tableData: [],
      pagination: pagination
    });
  }
}

function fetchTableData(_ref) {
  var _this = this;

  var str = _ref.str,
      current = _ref.current,
      pageSize = _ref.pageSize;
  var userRange = this.props.userRange;
  var currentPage = current - 1;
  var fyfjm = parseInt(userRange) === 3 && userInfo ? userInfo.fyfjm : "";
  var params = {
    queryStr: str,
    page: currentPage,
    size: pageSize,
    fyfjm: fyfjm
  };
  userSelect.findUserDTOPage(params).then(function (res) {
    return res.json();
  }).then(function (res) {
    if (res.code === "success" && res.data) {
      var pagination = _this.state.pagination;
      pagination.current = current;
      pagination.pageSize = pageSize;
      pagination.total = res.data.totalElements;
      var arr = [];
      res.data.content.forEach(function (item) {
        arr.push({
          id: item.userid,
          username: item.username,
          value: item.username,
          nickname: item.nickname,
          label: item.nickname,
          phone: item.phone,
          zj: item.zj,
          zw: item.zw,
          rybs: item.rybs,
          orgName: item.orgName,
          departmentName: item.departmentName,
          dfsId: item.dfsId,
          departmentInfo: (item.departmentName ? item.departmentName : "") + " / " + (item.orgName ? item.orgName : null),
          fyfjm: item.fyfjm,
          bmbs: item.bmbs
        });
      });

      _this.setState({
        tableData: arr,
        pagination: pagination
      });
    }
  });
}

;

function onClickList(id, username, nickname, rybs, orgName, departmentName, fyfjm, bmbs) {
  onClickSearchClear.call(this);
  var selectedRys = this.state.selectedRys;
  var filterData = selectedRys.filter(function (item) {
    return item.username == username;
  });

  if (filterData.length == 0) {
    selectedRys.push({
      id: id,
      username: username,
      nickname: nickname,
      rybs: rybs,
      orgName: orgName,
      departmentName: departmentName,
      fyfjm: fyfjm,
      bmbs: bmbs
    });
    this.setState({
      selectedRys: selectedRys
    });
  }
}

function onOkModal(selectedRys, onOk, onCancel) {
  var latelyContactors = selectedRys.concat();

  if (latelyContactors.length > latelyContactorsSize) {
    latelyContactors.splice(latelyContactorsSize);
    localStorage.setItem("latelyContactors", _JSON$stringify(latelyContactors));
  } else {
    var data = JSON.parse(localStorage.getItem("latelyContactors"));

    if (data) {
      var _loop = function _loop() {
        if (_isArray) {
          if (_i >= _iterator.length) return "break";
          _ref2 = _iterator[_i++];
        } else {
          _i = _iterator.next();
          if (_i.done) return "break";
          _ref2 = _i.value;
        }

        var item = _ref2;

        if (latelyContactors.length > latelyContactorsSize - 1) {
          return "break";
        }

        var ts = latelyContactors.filter(function (contactor) {
          return contactor.username == item.username;
        });

        if (ts.length == 0) {
          latelyContactors.push(item);
        }
      };

      for (var _iterator = data, _isArray = Array.isArray(_iterator), _i = 0, _iterator = _isArray ? _iterator : _getIterator(_iterator);;) {
        var _ref2;

        var _ret = _loop();

        if (_ret === "break") break;
      }
    }

    localStorage.setItem("latelyContactors", _JSON$stringify(latelyContactors));
  }

  onOk(selectedRys);
  onCancel();
}

var RySelectContent =
/*#__PURE__*/
function (_Component) {
  _inheritsLoose(RySelectContent, _Component);

  function RySelectContent(props) {
    var _this2;

    _this2 = _Component.call(this, props) || this;
    _this2.state = {
      selectedRys: [],
      tableData: [],
      pagination: {
        size: "small",
        showTotal: function showTotal(total, range) {
          return "\u603B\u8BA1 " + total + " \u6761";
        },
        current: 1,
        pageSize: 14,
        total: 0,
        onChange: function onChange(current, pageSize) {
          return fetchTableData.call(_assertThisInitialized(_assertThisInitialized(_this2)), {
            str: _this2.state.searchValue,
            current: current,
            pageSize: pageSize
          });
        }
      },
      searchValue: ""
    };
    return _this2;
  }

  var _proto = RySelectContent.prototype;

  _proto.componentDidMount = function componentDidMount() {
    if (this.props.defaultSelectedRys) {
      this.setState({
        selectedRys: this.props.defaultSelectedRys.concat()
      });
    }
  };

  _proto.componentWillReceiveProps = function componentWillReceiveProps(nextProps) {
    var _this3 = this;

    if (nextProps.defaultSelectedRys && this.props.defaultSelectedRys) {
      if (nextProps.defaultSelectedRys.filter(function (nextPropsItem) {
        return _this3.props.defaultSelectedRys.filter(function (thisPropsItem) {
          return thisPropsItem.username == nextPropsItem.username;
        }).length == 0;
      }).length > 0 || nextProps.defaultSelectedRys.length != this.props.defaultSelectedRys.length) {
        this.setState({
          selectedRys: nextProps.defaultSelectedRys.concat()
        });
      }
    }
  };

  _proto.render = function render() {
    var _this4 = this;

    return React.createElement("div", null, React.createElement(Row, {
      gutter: 32
    }, React.createElement(Col, {
      span: 16
    }, React.createElement(Search, {
      placeholder: "\u8F93\u5165\u5173\u952E\u5B57\u641C\u7D22",
      suffix: this.state.searchValue !== '' ? React.createElement(Icon, {
        type: "close-circle",
        onClick: function onClick() {
          return onClickSearchClear.call(_this4);
        }
      }) : null,
      value: this.state.searchValue,
      onChange: function onChange(e) {
        return _this4.setState({
          searchValue: e.target.value
        });
      },
      onDelayChange: function onDelayChange(value) {
        return onChangeSearch.call(_this4, value);
      }
    }), this.state.searchValue === "" ? React.createElement(Card, {
      style: {
        marginTop: 10
      },
      bodyStyle: {
        overflowY: "auto",
        height: 588
      }
    }, React.createElement(RyTree, {
      userRange: this.props.userRange,
      selectedRys: this.state.selectedRys,
      onChange: function onChange(selectedRys) {
        return _this4.setState({
          selectedRys: selectedRys
        });
      }
    })) : React.createElement("div", {
      style: {
        marginTop: 10
      }
    }, React.createElement(QueryList, {
      dataSource: this.state.tableData,
      pagination: this.state.pagination,
      onClick: function onClick(item) {
        return onClickList.call(_this4, item.id, item.username, item.nickname, item.rybs, item.orgName, item.departmentName, item.fyfjm, item.bmbs);
      }
    }))), React.createElement(Col, {
      span: 8
    }, React.createElement(Card, {
      title: "\u5DF2\u9009\u4EBA\u5458",
      extra: React.createElement("a", {
        href: "javascript:;",
        onClick: function onClick() {
          return _this4.setState({
            selectedRys: []
          });
        }
      }, "\u6E05\u7A7A"),
      className: stylePrefix + '-card',
      bodyStyle: {
        overflowY: "auto",
        height: 600,
        padding: 0
      }
    }, React.createElement(SelectedList, {
      selectedRys: this.state.selectedRys,
      onClose: function onClose(username) {
        return onCloseSelectedList.call(_this4, username);
      }
    })))), React.createElement("div", {
      style: {
        textAlign: 'right',
        marginTop: 16
      }
    }, React.createElement(Button, {
      style: {
        marginRight: 16
      },
      onClick: this.props.onCancel
    }, "\u53D6\u6D88"), React.createElement(Button, {
      type: "primary",
      onClick: function onClick() {
        return onOkModal.call(_this4, _this4.state.selectedRys, _this4.props.onOk, _this4.props.onCancel);
      }
    }, "\u786E\u5B9A")));
  };

  return RySelectContent;
}(Component);

RySelectContent.propTypes = {
  defaultSelectedRys: PropTypes.array,
  userRange: PropTypes.number,
  onOk: PropTypes.func,
  onCancel: PropTypes.func
};
RySelectContent.defaultProps = {
  userRange: 3
};
export default RySelectContent;
