// fay-react-ui/res/business/rySelector/selectedList.js
/**
 * Created by Administrator on 2018/5/21.
 */
import React from 'react';
import Icon from '../../../icon';
var stylePrefix = 'rj-rySelector-selectedList';
export default (function (_ref) {
  var selectedRys = _ref.selectedRys,
      onClose = _ref.onClose;

  var ItemContent = function ItemContent(_ref2) {
    var nickname = _ref2.nickname,
        username = _ref2.username;
    return React.createElement("div", {
      className: "item",
      title: nickname,
      onDoubleClick: function onDoubleClick() {
        return onClose(username);
      }
    }, React.createElement("div", {
      className: "content"
    }, nickname), React.createElement("div", {
      className: "close"
    }, React.createElement(Icon, {
      type: "close",
      onClick: function onClick() {
        return onClose(username);
      }
    })));
  };

  return React.createElement("div", {
    className: stylePrefix
  }, React.createElement("div", {
    className: "listBody"
  }, selectedRys.map(function (item) {
    return React.createElement(ItemContent, {
      key: item.username,
      username: item.username,
      nickname: item.nickname
    });
  })));
});
