// fay-react-ui/res/own/modal/content.js
import _inheritsLoose from "@babel/runtime-corejs2/helpers/inheritsLoose";
import React from 'react';
import Scroll from '../customScroll';

var Content =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(Content, _React$Component);

  function Content(props) {
    var _this;

    _this = _React$Component.call(this, props) || this;
    _this.modalContentRef = React.createRef();
    _this.state = {
      scrollWidth: 0
    };
    return _this;
  }

  var _proto = Content.prototype;

  _proto.componentDidMount = function componentDidMount() {
    var _this$modalContentRef = this.modalContentRef.current,
        offsetWidth = _this$modalContentRef.offsetWidth,
        offsetHeight = _this$modalContentRef.offsetHeight;
    this.setState({
      scrollWidth: offsetWidth + 17
    });
  };

  _proto.render = function render() {
    var _this$props = this.props,
        maxHeight = _this$props.maxHeight,
        minHeight = _this$props.minHeight,
        maxWidth = _this$props.maxWidth,
        minWidth = _this$props.minWidth,
        height = _this$props.height,
        children = _this$props.children;
    return React.createElement("div", {
      ref: this.modalContentRef,
      style: {
        width: '100%',
        height: '100%'
      }
    }, React.createElement(Scroll, {
      maxHeight: maxHeight,
      minHeight: minHeight,
      maxWidth: maxWidth,
      minWidth: minWidth,
      width: "100%",
      height: height
    }, children));
  };

  return Content;
}(React.Component);

export { Content as default };
