// fay-react-ui/res/business/rySelector/rySelectModal.js
import _extends from "@babel/runtime-corejs2/helpers/extends";
import _objectWithoutPropertiesLoose from "@babel/runtime-corejs2/helpers/objectWithoutPropertiesLoose";

/**
 * Created by Administrator on 2018/5/18.
 */
import React from "react";
import RySelectContent from './rySelectContent';
import "./style/index.less";
import Modal from "../../../modal";
var stylePrefix = 'rj-rySelector-rySelectModal';
export default (function (_ref) {
  var userRange = _ref.userRange,
      defaultSelectedRys = _ref.defaultSelectedRys,
      onOk = _ref.onOk,
      modalProps = _objectWithoutPropertiesLoose(_ref, ["userRange", "defaultSelectedRys", "onOk"]);

  return React.createElement(Modal, _extends({
    title: "\u4EBA\u5458\u9009\u62E9",
    footer: null,
    maskClosable: false,
    mask: false,
    width: 700,
    wrapClassName: stylePrefix
  }, modalProps), React.createElement(RySelectContent, {
    defaultSelectedRys: defaultSelectedRys,
    userRange: userRange,
    onOk: onOk,
    onCancel: modalProps.onCancel
  }));
});
