// fay-react-ui/res/business/rySelector/selectedInput.js
/**
 * Created by Administrator on 2018/5/25.
 */
import React from "react";
import QueryInput from "./queryInput";
import Tag from "../../../tag";
var stylePrefix = 'rj-rySelector-selectedInput';
export default (function (_ref) {
  var value = _ref.value,
      selectedRys = _ref.selectedRys,
      width = _ref.width,
      onChange = _ref.onChange,
      _onClose = _ref.onClose,
      onBackspace = _ref.onBackspace;
  return React.createElement("div", {
    className: stylePrefix,
    style: {
      width: width
    }
  }, React.createElement("div", {
    className: "ant-input"
  }, selectedRys.map(function (item) {
    return React.createElement("div", {
      key: item.username,
      style: {
        display: "inline-block"
      }
    }, React.createElement(Tag, {
      closable: true,
      onClose: function onClose(e) {
        return _onClose(item.username);
      }
    }, React.createElement("span", {
      className: "textOverflow",
      style: {
        width: 44,
        float: "left"
      }
    }, item.nickname)));
  }), React.createElement("div", {
    style: {
      display: "inline-block"
    }
  }, React.createElement("div", {
    style: {
      display: "inline-block"
    }
  }, React.createElement(QueryInput, {
    value: value,
    onChange: onChange,
    onBackspace: onBackspace
  })))));
});
