// fay-react-ui/res/business/error/e404/views/index.js
/**
 * Created by feichongzheng on 17/6/9.
 */
import React from 'react';
import Row from '../../../../../row';
import Col from '../../../../../col';
import { Link } from 'react-router-dom';
export default (function () {
  return React.createElement(Row, {
    style: {
      textAlign: 'center',
      paddingTop: '100px'
    }
  }, React.createElement(Col, {
    span: 5
  }, " "), React.createElement(Col, {
    span: 14
  }, React.createElement("div", {
    style: {
      fontSize: '8em'
    }
  }, "4", React.createElement("img", {
    src: "/resource/images/error/giphy.jpg",
    width: "80px",
    style: {
      border: '3px solid #000',
      borderRadius: '50px'
    }
  }), "4"), React.createElement("div", {
    style: {
      fontSize: '1.5em'
    }
  }, "\u975E\u5E38\u62B1\u6B49\uFF0C\u60A8\u8FF7\u8DEF\u4E86\uFF01"), React.createElement("div", {
    style: {
      fontSize: '1.5em'
    }
  }, "\u60A8\u8BBF\u95EE\u7684\u9875\u9762\u4E0D\u5B58\u5728\uFF01"), React.createElement("div", {
    style: {
      fontSize: '1.5em',
      marginTop: '50px'
    }
  }, React.createElement(Link, {
    to: "/"
  }, "\u8FD4\u56DE\u9996\u9875"))));
});
