// fay-react-ui/res/business/error/e401D3/views/index.js
/**
 * Created by feichongzheng on 17/6/9.
 */
import React from 'react';
import Row from '../../../../../row';
import Col from '../../../../../col';
import { Link } from 'react-router-dom';
export default (function () {
  return React.createElement(Row, {
    style: {
      textAlign: 'center',
      paddingTop: '100px'
    }
  }, React.createElement(Col, {
    span: 5
  }, " "), React.createElement(Col, {
    span: 14
  }, React.createElement("div", {
    style: {
      fontSize: '8em'
    }
  }, "4", React.createElement("img", {
    src: "/resource/images/error/giphy.jpg",
    width: "80px",
    style: {
      border: '3px solid #000',
      borderRadius: '50px'
    }
  }), "1"), React.createElement("div", {
    style: {
      fontSize: '1.5em'
    }
  }, "\u975E\u5E38\u62B1\u6B49\uFF0C\u60A8\u7684\u8BBF\u95EE\u53D7\u5230\u9650\u5236\uFF01"), React.createElement("div", {
    style: {
      fontSize: '1.5em'
    }
  }, "\u5982\u60F3\u8981\u8BBF\u95EE\uFF0C\u8BF7\u8054\u7CFB\u60A8\u7684\u7BA1\u7406\u5458\uFF01"), React.createElement("div", {
    style: {
      fontSize: '1.5em',
      marginTop: '50px'
    }
  }, React.createElement(Link, {
    to: "/"
  }, "\u8FD4\u56DE\u9996\u9875"))));
});
